lib/optimize/CommonsChunkPlugin.js

```diff
@@ -24,153 +24,304 @@ The available options are:
 	minSize: number`);
 		}
 
-		if(Array.isArray(options) || typeof options === "string") {
-			options = {
-				name: options
+		const normalizedOptions = this.normalizeOptions(options);
+
+		this.chunkNames = normalizedOptions.chunkNames;
+		this.filenameTemplate = this.filenameTemplate;
+		this.minChunks = normalizedOptions.minChunks;
+		this.selectedChunks = normalizedOptions.selectedChunks;
+		this.async = normalizedOptions.async;
+		this.minSize = normalizedOptions.minSize;
+		this.ident = normalizedOptions.ident;
```

Comment:
Maybe it's better to store the (normalized) options in `this.options` and just use the properties from this object in the apply method.